src/shader_preprocessor.ts

```diff
@@ -56,7 +56,7 @@ function mapToGlslTypes(type: DataType): GLSLDataType|DataType {
 
 interface ProgramParams {
   dispatchLayout: {x: number[], y?: number[], z?: number[]};
-  workGroupSize: [number, number, number];
+  workGroupSize?: [number, number, number];
```

Comment:
I made it as non-optional previously. If you want to change it to optional, please modify the comments at the beginning of this file:
Remove '// 2. Work group layout 'layout (local_size_x, ...)' is a MUST not optional.'  in 'The differences with webgpu backend'.

Reply:
Removed item 2